Index: source/padialog.cxx
===================================================================
RCS file: /cvs/gsl/padmin/source/padialog.cxx,v
retrieving revision 1.21
diff -u -r1.21 padialog.cxx
--- openoffice.org.orig/padmin/source/padialog.cxx	10 Jul 2006 16:39:26 -0000	1.21
+++ openoffice.org/padmin/source/padialog.cxx	24 Sep 2006 14:41:14 -0000
@@ -182,6 +182,7 @@
 	::psp::PrintFontManager& rFontManager( ::psp::PrintFontManager::get() );
 	if( ! rFontManager.checkImportPossible() )
 		m_aFontsPB.Enable( FALSE );
+	m_aFontsPB.Show(FALSE);
 }
 
 PADialog::~PADialog()
Index: source/text/shared/guide/spadmin.xhp
===================================================================
RCS file: /cvs/documentation/helpcontent2/source/text/shared/guide/spadmin.xhp,v
retrieving revision 1.9
diff -u -r1.9 spadmin.xhp
--- openoffice.org.orig/helpcontent2/source/text/shared/guide/spadmin.xhp	28 Jun 2006 15:43:29 -0000	1.9
+++ openoffice.org/helpcontent2/source/text/shared/guide/spadmin.xhp	24 Sep 2006 14:46:08 -0000
@@ -58,8 +58,6 @@
 <bookmark_value>PostScript; PDF converter, UNIX</bookmark_value>
 <bookmark_value>converters; PostScript, UNIX</bookmark_value>
 <bookmark_value>PDF; PostScript to PDF converter, UNIX</bookmark_value>
-<bookmark_value>fonts; adding under UNIX</bookmark_value>
-<bookmark_value>typefaces; adding under UNIX</bookmark_value>
 </bookmark><comment>mw deleted "printer programs under UNIX",  "fax programs under UNIX" and "font programs under UNIX" and changed "faxes;..."</comment>
 <paragraph role="paragraph" id="par_id3154422" xml-lang="en-US" l10n="E" oldref="341" localize="false"/>
 <paragraph role="heading" id="hd_id3147834" xml-lang="en-US" level="1" l10n="U" oldref="1"><variable id="spadmin"><link href="text/shared/guide/spadmin.xhp" name="Setting up Printer, Fax and Fonts Under UNIX Based Platforms">Setting up Printer, Fax and Fonts Under UNIX Based Platforms</link>
@@ -77,7 +75,6 @@
 </list>
 <paragraph role="paragraph" id="par_id3152349" xml-lang="en-US" l10n="U" oldref="14">After it starts, the window of the printer administration program <emph>spadmin</emph> appears.</paragraph>
 <paragraph role="paragraph" id="par_id3149580" xml-lang="en-US" l10n="U" oldref="10">Following a server installation, the system administrator first logs on as with root privileges, and starts the printer administration program <emph>spadmin</emph>. The administrator then creates a general printer configuration file called {install_path}/share/psprint/psprint.conf for all users. All changes are immediately available to all users.</paragraph>
-<paragraph role="paragraph" id="par_id3153822" xml-lang="en-US" l10n="U" oldref="285">The system administrator can also add fonts for all users in the server installation. However, these fonts are available only after restarting the $[officename] software.</paragraph>
 <paragraph role="heading" id="hd_id3159177" xml-lang="en-US" level="2" l10n="U" oldref="121">Setting up Printers</paragraph>
 <paragraph role="paragraph" id="par_id3159157" xml-lang="en-US" l10n="U" oldref="122">Under UNIX based platforms, the $[officename] software only offers direct support for printers using the PostScript technology. Other printers must be set up as described in the section <emph>Printer Drivers in the $[officename] Software</emph>. The $[officename] software automatically provides a printer with the default driver for each system queue. You can add additional printers as needed.</paragraph>
 <paragraph role="heading" id="hd_id3148564" xml-lang="en-US" level="3" l10n="U" oldref="286">Adding a Printer</paragraph>
@@ -234,88 +231,6 @@
 </list>
 <paragraph role="paragraph" id="par_id3153064" xml-lang="en-US" l10n="U" oldref="313">You can now create PDF documents by printing to the converter that has just been created.</paragraph>
 <paragraph role="paragraph" id="par_id3159222" xml-lang="en-US" l10n="E" oldref="345" localize="false"/>
-<paragraph role="heading" id="hd_id3156195" xml-lang="en-US" level="2" l10n="U" oldref="143">Installing Fonts</paragraph>
-<paragraph role="paragraph" id="par_id3153914" xml-lang="en-US" l10n="U" oldref="144">When you are working with the $[officename] software you might notice that a different number of fonts is provided depending on the document type being used. This is because not all of the fonts can be used in every case.</paragraph>
-<list type="unordered">
-<listitem>
-<paragraph role="listitem" id="par_id3152382" xml-lang="en-US" l10n="U" oldref="145">Therefore, only those fonts will be shown in the font selection box when working with a text document which can also be printed, as it is assumed that you will only want to use those fonts that you can actually have printed on paper.</paragraph>
-</listitem>
-<listitem>
-<paragraph role="listitem" id="par_id3157982" xml-lang="en-US" l10n="CHG" oldref="146">In an HTML document or in Web layout, only fonts that are available on screen are offered.</paragraph>
-</listitem>
-<listitem>
-<paragraph role="listitem" id="par_id3149590" xml-lang="en-US" l10n="U" oldref="147">On the other hand, when working with spreadsheets and drawings you will be able to use all of the fonts that can be either printed or shown on the screen.</paragraph>
-</listitem>
-</list>
-<paragraph role="paragraph" id="par_id3159093" xml-lang="en-US" l10n="U" oldref="148">The $[officename] software tries to match the display on screen to the printout (WYSIWYG = what you see is what you get). Possible problems are shown in the bottom of the dialog <emph>Format - Character</emph>.</paragraph>
-<paragraph role="heading" id="hd_id3156353" xml-lang="en-US" level="3" l10n="U" oldref="314">Adding Fonts</paragraph>
-<paragraph role="paragraph" id="par_id3144503" xml-lang="en-US" l10n="U" oldref="315">You can integrate additional fonts in the $[officename] software. Fonts which you integrate are available exclusively to the $[officename] software and can be used with various Xservers without your having to install them there. To make the fonts available to other programs as well, proceed as usual by adding the fonts to your Xserver. The $[officename] software can display and print out PostScript Type1 fonts as well as TrueType fonts (including TrueType Collections).</paragraph>
-<paragraph role="paragraph" id="par_id3156174" xml-lang="en-US" l10n="U" oldref="316">To integrate additional fonts in the $[officename] software, proceed as follows:</paragraph>
-<list type="ordered">
-<listitem>
-<paragraph role="listitem" id="par_id3154825" xml-lang="en-US" l10n="U" oldref="357">Go to the {install_path}/program directory.</paragraph>
-</listitem>
-<listitem>
-<paragraph role="listitem" id="par_id3151127" xml-lang="en-US" l10n="U" oldref="317">Enter: ./spadmin</paragraph>
-</listitem>
-<listitem>
-<paragraph role="listitem" id="par_id3149072" xml-lang="en-US" l10n="U" oldref="318">Click <emph>Fonts</emph>.</paragraph>
-</listitem>
-<listitem>
-<paragraph role="listitem" id="par_id3156341" xml-lang="en-US" l10n="U" oldref="319">The dialog lists all fonts added for the $[officename] software. You can select and remove fonts using the <emph>Remove</emph> button or add new fonts with the <emph>Add</emph> button.</paragraph>
-</listitem>
-<listitem>
-<paragraph role="listitem" id="par_id3155118" xml-lang="en-US" l10n="U" oldref="320">Click <emph>Add</emph>. The <emph>Add Fonts</emph> dialog appears.</paragraph>
-</listitem>
-<listitem>
-<paragraph role="listitem" id="par_id3149126" xml-lang="en-US" l10n="U" oldref="321">Enter the directory from which you want to add the fonts. Press the <emph>...</emph> button and select the directory from the path selection dialog or enter the directory directly.</paragraph>
-</listitem>
-<listitem>
-<paragraph role="listitem" id="par_id3154266" xml-lang="en-US" l10n="U" oldref="322">A list of the fonts from this directory appears. Select the fonts you want to add. To add all the fonts, click <emph>Select All</emph>.</paragraph>
-</listitem>
-<listitem>
-<paragraph role="listitem" id="par_id3154060" xml-lang="en-US" l10n="U" oldref="323">With the <emph>Create soft links only</emph> check box you can determine whether the fonts are to be copied into the $[officename] directory or only symbolic links are to be created there. If the fonts to be added are on a data medium that is not always available (such as a CD-ROM), you must copy the fonts.</paragraph>
-</listitem>
-<listitem>
-<paragraph role="listitem" id="par_id3155539" xml-lang="en-US" l10n="U" oldref="324">Click OK. The fonts will now be added.</paragraph>
-</listitem>
-</list>
-<paragraph role="paragraph" id="par_id3146776" xml-lang="en-US" l10n="U" oldref="325">In the case of a server installation, the fonts are installed in that installation if possible. If the user has no write access rights, the fonts will be installed in the corresponding user installation so that only the user who installed them can access them.</paragraph>
-<paragraph role="heading" id="hd_id3152869" xml-lang="en-US" level="3" l10n="U" oldref="326">Deleting Fonts</paragraph>
-<paragraph role="paragraph" id="par_id3151302" xml-lang="en-US" l10n="U" oldref="327">To delete fonts, proceed as follows:</paragraph>
-<list type="ordered">
-<listitem>
-<paragraph role="listitem" id="par_id3150213" xml-lang="en-US" l10n="U" oldref="328">Start spadmin.</paragraph>
-</listitem>
-<listitem>
-<paragraph role="listitem" id="par_id3154608" xml-lang="en-US" l10n="U" oldref="329">Click <emph>Fonts</emph>.</paragraph>
-</listitem>
-<listitem>
-<paragraph role="listitem" id="par_id3145195" xml-lang="en-US" l10n="U" oldref="330">All fonts added to the $[officename] software are listed in the dialog that now appears. Select the fonts you want to delete and click <emph>Delete</emph>.</paragraph>
-</listitem>
-</list>
-<paragraph role="paragraph" id="par_id3153242" xml-lang="en-US" l10n="U" oldref="331">You can delete only the fonts that have been added for the $[officename] software.</paragraph>
-<paragraph role="heading" id="hd_id3147487" xml-lang="en-US" level="3" l10n="U" oldref="332">Renaming Fonts</paragraph>
-<paragraph role="paragraph" id="par_id3148899" xml-lang="en-US" l10n="U" oldref="333">You can also rename fonts added for the $[officename] software. This is especially useful for fonts that contain several localized names.</paragraph>
-<list type="ordered">
-<listitem>
-<paragraph role="listitem" id="par_id3154766" xml-lang="en-US" l10n="U" oldref="334">Start spadmin.</paragraph>
-</listitem>
-<listitem>
-<paragraph role="listitem" id="par_id3151009" xml-lang="en-US" l10n="U" oldref="335">Click <emph>Fonts</emph>.</paragraph>
-</listitem>
-<listitem>
-<paragraph role="listitem" id="par_id3151120" xml-lang="en-US" l10n="U" oldref="336">Select the fonts you want to rename and click <emph>Rename</emph>.</paragraph>
-</listitem>
-<listitem>
-<paragraph role="listitem" id="par_id3153008" xml-lang="en-US" l10n="U" oldref="337">Enter a new name in the dialog that appears. If the font contains several names, these names will serve as suggestions in the combo box where you enter the new name.</paragraph>
-</listitem>
-<listitem>
-<paragraph role="listitem" id="par_id3155521" xml-lang="en-US" l10n="U" oldref="338">Click OK.</paragraph>
-</listitem>
-</list>
-<paragraph role="paragraph" id="par_id3149802" xml-lang="en-US" l10n="U" oldref="339">If you select several fonts to rename, one dialog appears for each selected font.</paragraph>
-<paragraph role="paragraph" id="par_id3153487" xml-lang="en-US" l10n="U" oldref="340">If you have selected a TrueType Collection (TTC), one dialog appears for each of the fonts contained in it.</paragraph>
 </section>
 </section>
 </body>
