--- svx.orig/source/svdraw/svdfppt.cxx	2010-05-19 09:10:44.000000000 +0100
+++ svx/source/svdraw/svdfppt.cxx	2010-05-19 09:28:25.000000000 +0100
@@ -7120,6 +7120,10 @@
 															PPTCharPropSet* pCurrent = (PPTCharPropSet*)aCharPropList.GetObject( nIdx );
 															sal_Int32		nNextStringLen = pCurrent->maString.Len();
 
+															DBG_ASSERT( pFE->pField1, "missing field!" );
+															if (!pFE->pField1)
+															    break;
+
 															const SvxURLField* pField = (const SvxURLField*)pFE->pField1->GetField();
 
 															if ( pCurrent->mpFieldItem )
