Index: svx/source/dialog/srchdlg.cxx
===================================================================
--- svx/source/dialog/srchdlg.cxx	(revision 272843)
+++ svx/source/dialog/srchdlg.cxx	(working copy)
@@ -873,6 +873,18 @@
         pMoreBtn->AddWindow( &aNoFormatBtn );
     }
 
+    if (bDrawApp || bImpressApp)
+    {
+        // "Find All" button is hidden--align "Find" vertically to the
+        // search listbox
+        Point aNewPt(aSearchBtn.GetPosPixel());
+        const Size aBtnSz(aSearchBtn.GetSizePixel());
+        const Size aLBSz(aSearchLB.GetSizePixel());
+        const int nOff((aLBSz.Height() - aBtnSz.Height()) / 2);
+        aNewPt.Y() = aSearchLB.GetPosPixel().Y() + nOff;
+        aSearchBtn.SetPosPixel(aNewPt);
+    }
+
     if ( bDrawApp )
     {
         // Draw App: "Regular expressions" and "Search for Styles" check boxes are hidden
@@ -1058,6 +1070,8 @@
 
 		if ( pSearchItem->GetAppFlag() == SVX_SEARCHAPP_DRAW )
 		{
+            aSearchAllBtn.Hide();
+
 			aRegExpBtn.Hide();
 			aLayoutBtn.Hide();
 
